libs/audio-recording/recording.ts

```diff
@@ -238,21 +232,50 @@ namespace record {
         }
     }
 
-    export function isEmpty(): boolean {
-        _init()
-        return _memoryFill <= 0
-    }
-
     /**
      * Change how sensitive the microphone is. This changes the recording quality!
      */
     //% block="set microphone sensitivity to $gain"
-    //% gain.defl=Medium
-    //% weight=40
-    export function setMicGain(gain: AudioGain): void {
-        _init()
+    //% blockId="record_setMicGain"
+    //% weight=30
+    export function setMicGain(gain: AudioLevels): void {
+        // _init()
         _micGain = gain
         setMicrophoneGain(gain);
         return
     }
+
+    /**
+     * Set the sample frequency for recording, playback, or both (default)
+     * 
+     * @param hz The sample frequency, in Hz
+     */
+    //% block="set sample rate to $hz || for $scope"
+    //% blockId="record_setSampleRate"
+    //% hz.min=1000 hz.max=22000 hz.defl=11000
+    //% expandableArgumentMode="enabled"
+    //% weight=40
+    export function setSampleRate(hz: number, scope?: AudioSampleRateScope): void {
+        // _init()
+        switch (scope) {
+            case AudioSampleRateScope.Everything:
+                setInputSampleRate(hz);
+                setOutputSampleRate(hz);
+                break;
+
+            case AudioSampleRateScope.Playback:
+                setOutputSampleRate(hz);
+                break;
+
+            case AudioSampleRateScope.Recording:
+                setInputSampleRate(hz);
+                break;
+
```

Comment:
You could put the `Everything` case here and then just have it fall through to the default case. Then just remove it from the beginning of the switch statement. This would make it clearer that the default is the `Everything` case.
```suggestion
case AudioSampleRateScope.Everything:
```